android-kotlin/app/src/main/java/com/mongodb/app/LoginActivity.kt

```diff
@@ -0,0 +1,108 @@
+package com.mongodb.app
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import io.realm.mongodb.Credentials
+/*
+* Launched whenever a user isn't already logged in. Allows a user to enter email
+* and password credentials to log in to an existing account or create a new account.
+*/
+class LoginActivity : AppCompatActivity() {
+    private lateinit var username: EditText
+    private lateinit var password: EditText
+    private lateinit var loginButton: Button
+    private lateinit var accountExistsToggleButton: Button
+    private var isCreateUserMode = true
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        username = findViewById(R.id.input_username)
+        password = findViewById(R.id.input_password)
+
+        loginButton = findViewById(R.id.button_create_login)
+        loginButton.setOnClickListener { onLoginButtonClicked() }
+        loginButton.isEnabled = true
+
+        accountExistsToggleButton = findViewById(R.id.button_existing_account)
+        accountExistsToggleButton.setOnClickListener { toggleCreateUserMode() }
+        accountExistsToggleButton.isEnabled = true
+    }
+
+    /**
+     *  Handle account creation and user authentication
+     */
+    private fun onLoginButtonClicked() {
+        val username = this.username.text.toString()
+        val password = this.password.text.toString()
+        if (isCreateUserMode) {
+            registerUser(username, password)
+        } else {
+           login(username, password)
```

Comment:
Is this indented correctly?
```suggestion
            login(username, password)
```

Reply:
Nitpick: `logIn()` would be correct. "Log in" is the verb, "login" is the noun.

Reply:
fixed